aws/rust-runtime/aws-http/src/content_encoding.rs

```diff
@@ -143,44 +137,57 @@ fn get_unsigned_chunk_bytes_length(payload_length: u64) -> u64 {
     hex_repr_len + CRLF.len() as u64 + payload_length + CRLF.len() as u64
 }
 
-fn trailers_as_aws_chunked_bytes(trailer_map: Option<HeaderMap>) -> Bytes {
-    use std::fmt::Write;
-
-    let mut trailers = String::new();
-    let mut already_wrote_first_trailer = false;
-
+/// Writes trailers out into a `string` and then converts that `String` to a `Bytes` before
+/// returning.
+///
+/// - Trailer names are separated by a single colon only, no space.
+/// - Trailer names with multiple values will be written out one line per value, with the name
+///   appearing on each line.
+fn trailers_as_aws_chunked_bytes(trailer_map: Option<HeaderMap>) -> BytesMut {
     if let Some(trailer_map) = trailer_map {
-        for (header_name, header_value) in trailer_map.into_iter() {
-            match header_name {
-                // New name, new value
-                Some(header_name) => {
-                    if already_wrote_first_trailer {
-                        // First trailer shouldn't have a preceding CRLF, but every trailer after it should
-                        trailers.write_str(CRLF).unwrap();
-                    } else {
-                        already_wrote_first_trailer = true;
-                    }
+        let mut current_header_name = None;
+        let mut trailers = BytesMut::new();
```

Comment:
Not blocking: Since we know the length of buffer needed for trailers (`total_rendered_length_of_trailers + 1`), you might consider using [`reserve`](https://docs.rs/bytes/1.1.0/bytes/struct.BytesMut.html#method.reserve) to reduce allocations.

Reply:
I used `with_capacity`, thanks for suggesting